tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/ProgrammeMembershipValidator.java

```diff
@@ -48,20 +49,23 @@ public ProgrammeMembershipValidator(PersonRepository personRepository,
    * based validation already in place. It checks that the person, programme and curriculum
    * entered.
    *
-   * @param programmeMembershipDTO the programmeMembership to check
+   * @param programmeMembershipDto the programmeMembership to check
+   * @return
    * @throws MethodArgumentNotValidException if there are validation errors
    */
-  public void validate(ProgrammeMembershipDTO programmeMembershipDTO)
+  public void validate(ProgrammeMembershipDTO programmeMembershipDto)
       throws MethodArgumentNotValidException {
 
+
     List<FieldError> fieldErrors = new ArrayList<>();
-    fieldErrors.addAll(checkPerson(programmeMembershipDTO));
-    fieldErrors.addAll(checkProgramme(programmeMembershipDTO));
-    fieldErrors.addAll(checkCurriculum(programmeMembershipDTO));
-    fieldErrors.addAll(checkRotation(programmeMembershipDTO));
+    fieldErrors.addAll(checkPerson(programmeMembershipDto));
+    fieldErrors.addAll(checkProgramme(programmeMembershipDto));
+    fieldErrors.addAll(checkCurriculum(programmeMembershipDto));
+    fieldErrors.addAll(checkRotation(programmeMembershipDto));
+    checkProgrammeDates(fieldErrors, programmeMembershipDto);
```

Comment:
Is there a reason why the checkProgrammeDates doesn't follow the same pattern as the others (checkRotation etc) and return any new fieldError entries? It's a minor point, but it might be worth tweaking just to keep these functions all looking the same. Otherwise it all looks good to me!